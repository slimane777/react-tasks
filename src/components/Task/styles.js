import styled from 'styled-components';

export const InputWrapper = styled.div`
    display: flex;
    padding: 20px;
    margin-top: 5px;
    border-bottom: 1px solid #80808029;
`;

export const Container = styled.div`
    border-radius: 8px;
    background-color: white;
    max-width: 500px;
    margin: auto;
    margin-top: 100px;
    overflow: hidden
`;

export const Input = styled.input`
    color: #8683a0;
    border: none;
    font-size: 1.1em;
    font-weight: 700;
    outline: none;
    ::placeholder { 
        color: #8683a0;
        opacity: 0.8; 
        font-weight: 600;
    };
    height: 30px;
    flex: 1
`;

export const Button = styled.button`
    background-color: transparent;
    border: none
`;