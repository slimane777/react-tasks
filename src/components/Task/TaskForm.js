import React from 'react'
import PropTypes from 'prop-types'
import { useSelector, useDispatch } from 'react-redux'
import {
  SET_TASK_TITLE_REQUESTED,
  CREATE_TASK_REQUESTED
} from '../../redux/actions/tasks'
import { InputWrapper, Input, Button } from './styles'
import { FiPlus, FiCheck } from "react-icons/fi";

const TaskForm = () => {

    const title = useSelector((state) => state.title);

    const dispatch = useDispatch();

    const setTaskTitle = (title) => dispatch({ type: SET_TASK_TITLE_REQUESTED, payload: title });
    const createTask = (title) => dispatch({ type: CREATE_TASK_REQUESTED, payload: title });

  const onChange = (e) => {
    setTaskTitle(e.target.value)
  }

  const onSubmit = (e) => {
    e.preventDefault()
    createTask(title)
  }

  return (
    
      <form onSubmit={onSubmit}>
        <InputWrapper>
          <FiPlus size={30} color={'#8683a0'} />
          <Input 
            type="text" 
            placeholder="Type your task" 
            onChange={onChange}
            value={title}
          />
          {
            !!title && <Button type="submit">
            <FiCheck size={30} color={'#3dd272'} />
          </Button>
          }
          
        </InputWrapper>
      </form>
  )
}

TaskForm.propTypes = {
  title: PropTypes.string,
//   setTaskTitle: PropTypes.func.isRequired,
//   createTask: PropTypes.func.isRequired
}

export default TaskForm;