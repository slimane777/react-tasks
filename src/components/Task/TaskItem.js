import React from 'react'
import PropTypes from 'prop-types'
import { TaskRow, TaskCheckBox, TaskLabel, TaskTime, CheckboxContainer, HiddenCheckbox, StyledCheckbox, Icon } from './styles'
import { useDispatch } from 'react-redux'
import moment from 'moment'
import {
  UPDATE_TASK_REQUESTED
} from '../../redux/actions/tasks'

const TaskItem = ({ task }) => {

  const dispatch = useDispatch()

  const updateTask = (taskObj) => dispatch({ type: UPDATE_TASK_REQUESTED, payload: taskObj });

  return (
    // <TaskRow onClick={()=>updateTask({id: task.id, title: task.title, completed: !task.completed, dateOfComplete: moment(new Date())})}>
    <TaskRow 
      onClick={(e)=>{
        console.log({e})
        console.log({task})
        e.preventDefault()
        let obj = {
          id: task.id, 
          title: task.title, 
          completed: !task.completed, 
          dateOfComplete: moment(new Date())
        }
        console.log({obj})
        updateTask(obj)
      }}
    >
      {/* <TaskCheckBox 
          type="checkbox"
          value={task.title}
          checked={task.completed}
          onChange={()=>dispatch()}
      /> */}

      <CheckboxContainer >
        {/* <HiddenCheckbox checked={task.completed} /> */}
        <StyledCheckbox 
          checked={task.completed ? task.completed : false} 
          // onChange={(e)=>{
          //   console.log({e})
          //   // let obj = {
          //   //   id: task.id, 
          //   //   title: task.title, 
          //   //   completed: !task.completed, 
          //   //   dateOfComplete: moment(new Date())
          //   // }
          //   // updateTask(obj)
          // }}
        >
          {
            task.completed &&
            <Icon viewBox="0 0 24 24">
              <polyline points="20 6 9 17 4 12" />
            </Icon>
          }
        </StyledCheckbox>
      </CheckboxContainer>

      <TaskLabel checked={task.completed}>
        {task.title}
      </TaskLabel>
      <TaskTime>
        {moment(new Date()).format('h:MM a')}
      </TaskTime>
     </TaskRow>
  )
}

TaskItem.propTypes = {
  task: PropTypes.object.isRequired,
}

export default TaskItem