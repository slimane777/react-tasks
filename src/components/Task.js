import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { useDispatch, useSelector } from 'react-redux'
import {
  GET_TASKS_REQUESTED,
  DELETE_TASK_REQUESTED
} from '../redux/actions/tasks';
// Components
import TaskForm from './TaskForm';
import TaskItem from './TaskItem';
import ClearButton from './Button/ClearButton';
import HeaderDate from './Date/HeaderDate';

const Task = () => {

    const tasks = useSelector((state) => state.tasks);
    
    const dispatch = useDispatch();

    useEffect(() => {
        console.log({tasks})
        getTasks()
    }, [])

    const getTasks = () => dispatch({ type: GET_TASKS_REQUESTED })

    const deleteTask = (id) => dispatch({ type: DELETE_TASK_REQUESTED, payload: id })

    return (
        <>
            <HeaderDate/>
            <ClearButton clearTasks={()=>{}} count={4} />
            <TaskForm />
            {tasks.loading && 'Loading...'}
            {tasks && tasks.tasks?.map((task, index) => (
                <TaskItem task={task} key={index} deleteTask={deleteTask} />
            ))}
        </>
    )
}

Task.propTypes = {
  loading: PropTypes.bool,
  tasks: PropTypes.array,
//   getTasks: PropTypes.func.isRequired,
//   deleteTask: PropTypes.func.isRequired
}

export default Task