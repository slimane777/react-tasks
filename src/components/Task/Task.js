import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import {
  GET_TASKS_REQUESTED,
  DELETE_TASK_REQUESTED
} from '../../redux/actions/tasks';
// Components
import TaskForm from './TaskForm';
import TaskItem from './TaskItem';
import Header from '../Header/Header';
import Loading from '../Loader/Loader';
import { Container } from './styles'

function Task() {

    const tasks = useSelector((state) => state.tasks);
    
    const dispatch = useDispatch();

    useEffect(() => {
        const getTasks = () => dispatch({ type: GET_TASKS_REQUESTED })
        getTasks()
    }, [])

    function compare( a, b ) {
        if ( a.id < b.id ){
          return -1;
        }
        if ( a.id > b.id ){
          return 1;
        }
        return 0;
    }

    return (
        <Container>
            <Header />
            <TaskForm />
            {tasks.loading && <Loading />}
            {tasks && tasks?.tasks?.sort(compare)?.map((task, index) => (
                <TaskItem task={task} key={index} />
            ))}
        </Container>
    )
}

export default Task