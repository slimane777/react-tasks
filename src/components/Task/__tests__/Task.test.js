import React from 'react';
import renderer from 'react-test-renderer';
import Task from '../Task';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

describe('Task test', () => {
  const initialState = {tasks:{tasks: []}}
  const mockStore = configureStore()
  let store

  store = mockStore(initialState)

  it('Task should match snapshot', () => {
    const component = renderer.create(<Provider store={store}><Task /></Provider>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()

  });

});
