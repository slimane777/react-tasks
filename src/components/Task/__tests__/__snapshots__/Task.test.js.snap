// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task test Task should match snapshot 1`] = `
<div
  className="sc-gtsrHT kRtws"
>
  <div
    className="sc-jrsJWt cwRHXg"
  >
    <div
      className="sc-hBMUJo hTiYPD"
    >
      <div
        className="sc-bqGGPW fogQeo"
      >
        <p
          className="sc-dIsUp gGxXqc"
        >
          Tuesday,
        </p>
        <p
          className="sc-dIsUp hDZOKa"
        >
           13th
        </p>
      </div>
      <p
        className="sc-ksluID jPhkaV"
      >
        July
      </p>
    </div>
    <div
      className="sc-crzoAE kNGeWW"
    >
      <span
        className="sc-iqAclL ctYqig"
      >
        0
         Tasks
      </span>
      <p
        className="sc-kEqXSa iWmGyP clear"
        onClick={[Function]}
      >
        Clear List
      </p>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="sc-bdnxRM kWHvNM"
    >
      <svg
        color="#c0bad2"
        fill="none"
        height={28}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": "#c0bad2",
          }
        }
        viewBox="0 0 24 24"
        width={28}
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="12"
          x2="12"
          y1="5"
          y2="19"
        />
        <line
          x1="5"
          x2="19"
          y1="12"
          y2="12"
        />
      </svg>
      <input
        className="sc-dlnjwi jxNCaK"
        onChange={[Function]}
        placeholder="Type your task"
        type="text"
      />
    </div>
  </form>
</div>
`;
